src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -378,7 +385,11 @@ static DIR *open_directory_secure_cwd(const char* dirname, int *saved_cwd_fd) {
     // Directory doesn't exist or is insecure, so there is nothing to cleanup.
     return dirp;
   }
+#if defined(_AIX)
+  int fd = dirp->dd_fd;
+#else
   int fd = dirfd(dirp);
+#endif
```

Comment:
I think this would be cleaner with AIX_ONLY/NOT_AIX macros.

Reply:
done